drizzlepac/hlautils/diagnostic_utils.py

```diff
@@ -0,0 +1,438 @@
+        hap product object specified by the "hap_obj" input.
+        data_source : string
+            name of the script that generated the data that will be stored in the "data" section
+
+        Returns
+        -------
+        Nothing.
+        """
+
+        if hap_obj and header_fits_filename:
+            errmsg = "Invalid inputs. Either 'hap_obj' OR 'header_fits_filename' needs to be specified. NOT BOTH."
+            log.error(errmsg)
+            raise Exception(errmsg)
+        elif not hap_obj and not header_fits_filename:
+            errmsg = "Invalid inputs. Either 'hap_obj' OR 'header_fits_filename' needs to be specified."
+            log.error(errmsg)
+            raise Exception(errmsg)
+        else:
+            if hap_obj:  # populate json header with HAP product information
+                self.header = hap_obj.primary_header
+                if hasattr(hap_obj,"filters"):
+                    self.filter = hap_obj.filter
+                else:
+                    if self.header['INSTRUME'].lower() == "acs":
+                        self.filter = poller_utils.determine_filter_name("{};{}".format(self.header['FILTER1'],self.header['FILTER2']))
+                    elif self.header['INSTRUME'].lower() == "wfc3":
+                        self.filter = poller_utils.determine_filter_name(self.header['FILTER'])
+                    else:
+                        errmsg = "Invalid instrument."
+                        log.error(errmsg)
+                        raise Exception(errmsg)
+            else:  # populate json header with HAP product information
+                if os.path.exists(header_fits_filename):
+                    self.header = getheader(header_fits_filename)
+                    if self.header['INSTRUME'].lower() == "acs":
+                        self.filter = poller_utils.determine_filter_name("{};{}".format(self.header['FILTER1'],self.header['FILTER2']))
+                    elif self.header['INSTRUME'].lower() == "wfc3":
+                        self.filter = poller_utils.determine_filter_name(self.header['FILTER'])
+                    else:
+                        errmsg = "Invalid instrument."
+                        log.error(errmsg)
+                        raise Exception(errmsg)
+                else:
+                    errmsg = "Invalid input. File {} does not exist.".format(header_fits_filename)
+                    log.error(errmsg)
+                    raise Exception(errmsg)
+
+
+        # gobble up inputs
+
+        self.data_source = data_source
+        self.description = description
+        # self.target_name = ""  # TODO: add header info: target name
+        # self.target_descrip = ""  # TODO: add header info: target description/category
+        # self.fgslock =''
+        # set logging level
+        log.setLevel(log_level)
+
+        # instantiate data storage dictionary
+        self._instantiate()
```

Comment:
Since you are creating an attribute in this private method (i.e., the variable is not already defined), please give the variable name here (self.out_dict).

Reply:
Not applicable.